astropy/modeling/physical_models.py

```diff
@@ -371,3 +372,334 @@ def input_units(self):
     def _parameter_units_for_data_units(self, inputs_unit, outputs_unit):
         return {'mass': outputs_unit[self.outputs[0]] * inputs_unit[self.inputs[0]] ** 3,
                 'r_plum': inputs_unit[self.inputs[0]]}
+
+
+class NFW(Fittable1DModel):
+    r"""
+    Navarro–Frenk–White (NFW) profile - model for radial distribution of dark matter.
+
+    Parameters
+    ----------
+    mass : float or :class:`~astropy.units.Quantity`
+        Mass of NFW peak within specified overdensity radius.
+    concentration : float
+        Concentration of the NFW profile.
+    redshift : float
+        Redshift of the NFW profile.
+    massfactor : tuple or str
+        Mass overdensity factor and type for provided profiles:
+            Tuple version:
+                ("virial") : virial radius
+
+                ("critical", N)  : radius where density is N times that of the critical density
+
+                ("mean", N)  : radius where density is N times that of the mean density
+
+            String version:
+                "virial" : virial radius
+
+                "Nc"  : radius where density is N times that of the critical density (e.g. "200c")
+
+                "Nm"  : radius where density is N times that of the mean density (e.g. "500m")
+    cosmo : :class:`~astropy.cosmology`
+        Background cosmology for density calculation.
+
+    Notes
+    -----
+
+    Model formula:
+
+    .. math:: \rho(r)=\frac{\delta_c\rho_{c}}{r/r_s(1+r/r_s)^2}
+
+    References
+    ----------
+    .. [1] https://arxiv.org/pdf/astro-ph/9508025
+    .. [2] https://en.wikipedia.org/wiki/Navarro%E2%80%93Frenk%E2%80%93White_profile
+    .. [3] https://en.wikipedia.org/wiki/Virial_mass
+    """
+
+    # Model Parameters
+
+    # NFW Profile mass and mass type
```

Comment:
If you move the comments below the attribute definitions they will appear in the documentation.

Reply:
Discussed and resolved offline